metaflow/plugins/kfp/tests/flows/s3_sensor_flow.py

```diff
@@ -0,0 +1,38 @@
+from metaflow import FlowSpec, step, resources, s3_sensor, Parameter
+
+def formatter(path: str, flow_parameters_json: dict) -> str:
+    return path.replace("CODE_PACKAGE_ID", flow_parameters_json["code_package_id"])
+
+@s3_sensor(
+    path="s3://serve-datalake-zillowgroup/zillow/workflow_sdk/metaflow_28d/dev/aip-integration-testing/HelloFlow/data/6f/CODE_PACKAGE_ID",
+    timeout_seconds=45,
+    polling_interval_seconds=1,
+    formatter=formatter
+)
+class S3SensorFlow(FlowSpec):
+    
+    # This parameter is just for testing @s3_sensor. In an actual flow,
+    # it doesn't make sense to pass the code package's metadata
+    # as a parameter, as that is transparently done behind the scenes.
+    alpha = Parameter(
+        "code_package_id",
+        default="6ff80f2870922f04ed6960dba2f23c7223e699a7",
+    )
```

Comment:
what are you testing or validating here? That `code_package_id` is in `flow_parameters_json` and `flow_parameters_json` is sent to the `formatter`?

Reply:
Correct; that we are able to use parameters in `flow_parameters_json` within `formatter`.